### ISSUE TITLE
Action not triggered when it's called from an HTML form action

### ISSUE THREAD

**Issue by romanstetsyk** [withastro/astro#12374]
### Astro Info

```block
Astro                    v4.16.9
Node                     v22.9.0
System                   macOS (x64)
Package Manager          npm
Output                   server
Adapter                  @astrojs/node
Integrations             none
```

### Describe the Bug

When calling an action from an HTML form action, as described [in the docs](https://docs.astro.build/en/guides/actions/#call-actions-from-an-html-form-action), the validation and handler execution are completely skipped if the following conditions are met:
1. `zod` schema has a `min` set.
2. The submitted field is not empty.
3. The length of the submitted value is shorter than `min` from the schema.

For example, in the repo linked below, enter a single character in the `password` field (it has the `min(8)` set in zod schema) and leave all other fields untouched.

In this case, `Astro.getActionResult` returns `undefined`.

Please note that I couldn't set up an example either in StackBlitz or in CodeSandbox. They don't seem to support calling actions from HTML form action.

Related issue: #11681

### What's the expected result?

The zod schema should validate the inputs, and if a field doesn’t meet the minimum length, it should throw a validation error.

### Link to Minimal Reproducible Example

https://github.com/romanstetsyk/astro-form-action

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by TheOtterlord** [comment]
Using the reproduction locally, I see `undefined` when fetching the page (as expected since no action is run). If I submit the form with an invalid length (4 characters), it correctly logs the error `ActionInputError: Failed to validate` and doesn't run the handler (as input is invalid). Tested in both dev and preview.

**Comment by romanstetsyk** [comment]
@TheOtterlord Thanks for the reply! I wanted to clarify that the form has four fields, and if you fill the password fields with 4 characters and leave other fields untouched, `Astro.getActionResult` will return `undefined`. It isn't the correct behavior. However, if you fill in any fields besides the password, the result is indeed the `AstroActionInputError`.

https://github.com/user-attachments/assets/7343e0e0-9b5f-42d3-853f-784a2bc772ce

**Comment by TheOtterlord** [comment]
Thanks. Managed to replicate entering one password. This was fixed in 5.0.0-beta.2 (see https://github.com/withastro/astro/pull/12016), so after upgrading to latest, the issue should be fixed.
